taskcluster/ac_taskgraph/transforms/ui_tests.py

```diff
@@ -0,0 +1,52 @@
+from taskgraph.transforms.base import TransformSequence
+
+
+transforms = TransformSequence()
+
+
+_ANDROID_TASK_NAME_PREFIX = "android-"
+
+
+@transforms.add
+def set_component_attribute(config, tasks):
+    for task in tasks:
+        component_name = task.pop("component", None)
+        if not component_name:
+            task_name = task["name"]
+            if task_name.startswith(_ANDROID_TASK_NAME_PREFIX):
+                component_name = task_name[len(_ANDROID_TASK_NAME_PREFIX):]
+            else:
+                raise NotImplementedError(f"Cannot determine component name from task {task_name}")
+
+        attributes = task.setdefault("attributes", {})
+        attributes["component"] = component_name
+
+        yield task
+
+
+@transforms.add
+def define_ui_test_command_line(config, tasks):
+    for task in tasks:
+        run = task.setdefault("run", {})
+        post_gradlew = run.setdefault("post-gradlew", [])
+        post_gradlew.append(
+            ['automation/taskcluster/androidTest/ui-test.sh', task["attributes"]["component"], 'arm', '1']
```

Comment:
We have something in our backlog for a while to make this more robust without needing additional parameters like the arch and shard count here (legacy decision 😄 ) but for the time-being this is fine.

Reply:
Sounds good! I'm happy to help make this code more generics once you all feel ready 🙂